data/contributor_data_template.json

```diff
@@ -1,92 +1,47 @@
 {
-    "stephensonc": {
-        "commits": [
-        ],
-        "issues_opened": [
-        ],
-        "issues_commented": [
-        ],
-        "pull_requests_opened": [
-        ],
-        "pull_requests_commented": [
-        ]
-    },
-    "koscinskic": {
-        "commits": [
-        ],
-        "issues_opened": [
-        ],
-        "issues_commented": [
-        ],
-        "pull_requests_opened": [
-        ],
-        "pull_requests_commented": [
-        ]
-    },
-    "schultzh": {
-        "commits": [
-        ],
-        "issues_opened": [
-        ],
-        "issues_commented": [
-        ],
-        "pull_requests_opened": [
-        ],
-        "pull_requests_commented": [
-        ]
-    },
     "RAW_DATA": [
-      {
-        "hash": "",
-        "author_msg": "",
-        "author_name": "",
-        "author_email": "",
-        "merge": false,
-        "line_added": 0,
-        "line_removed": 0,
-        "lines_of_code": 0,
-        "complexity": 0,
-        "methods": [],
-        "filename": [],
-        "filepath": []
-      },
-      {
-        "hash": "",
-        "author_msg": "",
-        "author_name": "",
-        "author_email": "",
-        "merge": true,
-        "line_added": 0,
-        "line_removed": 0,
-        "lines_of_code": 0,
-        "complexity": 0,
-        "methods": [],
-        "filename": [],
-        "filepath": []
-      }
-    ],
-    "INDIVIDUAL_METRICS": {
-        "schultzh": {
-            "EMAIL": "",
-            "COMMITS": 0,
-            "ADDED": 0,
-            "REMOVED": 0,
-            "TOTAL": 0,
-            "MODIFIED": 0,
-            "RATIO": 0,
-            "FILES": [],
-            "FORMAT": []
+        {
+            "hash": "",
+            "author_msg": "",
+            "author_name": "",
+            "author_email": "",
+            "merge": false,
+            "line_added": 0,
+            "line_removed": 0,
+            "lines_of_code": 0,
+            "complexity": 0,
+            "methods": [],
+            "filename": [],
+            "filepath": []
         },
-        "WonjoonC": {
-            "EMAIL": "",
-            "COMMITS": 0,
-            "ADDED": 0,
-            "REMOVED": 0,
-            "TOTAL": 0,
-            "MODIFIED": 0,
-            "RATIO": 0,
-            "FILES": [],
-            "FORMAT": []
+        {
+            "hash": "",
+            "author_msg": "",
+            "author_name": "",
+            "author_email": "",
+            "merge": true,
+            "line_added": 0,
+            "line_removed": 0,
+            "lines_of_code": 0,
+            "complexity": 0,
+            "methods": [],
+            "filename": [],
+            "filepath": []
         }
+    ],
+    "AUTHOR_USERNAME": {
```

Comment:
Is each username going to be a top level dictionary like `"RAW_DATA"?` If so, it seems like an inconsistency in dictionary levels. I'd propose that the top level dictionary for usernames would be something akin to `"DATA_BY_USERNAME"` instead and then the `"AUTHOR_USERNAME"` dictionary that is currently here would be moved to under the top level directory. This change would make the data access more consistent in that you would know which type of data you are accessing instead of having numerous top level keys with different organizational types.

Reply:
In our latest merge to master, we created two json files to avoid this issue. In the raw data storing file, the top-level can be accessed by the key `"RAW_DATA"` and it gives a list of all commits, on the metrics storing file, usernames are the top level. Since this is a template and I did not want to create two different template files, I merged both structures in one template. This will not affect the program in any way considering that it will be removed from master when the project is finalized.

Reply:
I was under the impression that the template file would stay in master to inform anyone who attempted to expand our project of the anticipated data storage structure that they would be interacting with. Has this been changed?

Reply:
I'm not sure, to be honest, I assumed that it was used to display the information to the architects' group. In the case that we are keeping it, how do you suggest I clarify the two different json layouts?

Reply:
I think that this can be merged as is, considering the impending due date of the project.